### PR TITLE
JUnit Testing Fixes

### PR DESCRIPTION
**Pull request by supertick**
* Service.java - protected ServiceConfig config member added and setName(name) removed isPeerStarted(key) added
* Runtime - release in reverse order, releaseAll with blocking/nonblocking and save runtime options
    reverse order most likely to be clean
    non-blocking - prevent deadlock issue
* Virtual MrlComm.java - fixed npe issues
* Arduino detach - if unconnected quick detach, do not wait for acks on msgs that will never get to mrlcomm
* Arduino - better checks for NPEs
* Arduino - re-entrant neoPixelAttach (previously added 2 when called on the same neopixel)
* AbstractServo - removed stopService - implemented disabling, detaching details in releaseService
* config removed from DeviceMapping (in Arduino)
* Jenkins - skipAfterFailureCount option added to verify
* ArduinoMsgGenerator - VirtualMsgs and MrlComm.java not handling msgs correctly due to invoke switch not set appropriately
* MAX_MSG_SIZE actually checked on Arduino --> MrlComm and dumped with error vs previously sending and the garbage it created
* More NPE checking in Arduino
* MrlNeoPixel virtual device cleaned up
* DiscordBot setBotName getBotName
* Servo - if autoDisable
* WebGui - removed mDNS chatty logging default

### CODE REVIEW

File: src/main/java/org/myrobotlab/service/Log.java [hunk 1]
```diff
@@ -254,6 +254,10 @@ public void setRootLogLevel(String level) {
     getLogLevel();
     broadcastState();
   }
+  
+  @Override
+  public void setName(String name) {
```
Comment (kwatters) [inline src/main/java/org/myrobotlab/service/Log.java:259]: why is this here?

File: src/main/java/org/myrobotlab/service/InMoov2Head.java [hunk 2]
```diff
@@ -502,15 +472,12 @@ public void test() {
...
-    }
-    if (param) {
-      blinkEyesTimer = new Timer();
-      new blinkEyesTimertask().run();
-    }
+  /**
+   * FIXME - implement
+   * @param b
+   */
+  public void autoBlink(boolean b) {
```
Comment (kwatters) [inline src/main/java/org/myrobotlab/service/InMoov2Head.java:479]: no functionality anymore..

File: src/main/java/org/myrobotlab/service/Runtime.java [hunk 3]
```diff
@@ -1424,17 +1450,17 @@ public static void shutdown() {
...
+
+      // release
       releaseAll();
     } catch (Exception e) {
       log.error("something threw - continuing to shutdown", e);
     }
 
+    // calling System.exit(0) before some specialized threads
+    // are completed will actually end up in a deadlock
+    Service.sleep(1000);
```
Comment (kwatters) [inline src/main/java/org/myrobotlab/service/Runtime.java:1463]: This seems bad.. why would system.exit(0) cause a deadlock?

File: src/main/resources/resource/WebGui/app/views/tabsViewCtrl.js [hunk 4]
```diff
@@ -33,12 +33,12 @@ angular.module('mrlapp.mrl').controller('tabsViewCtrl', ['$location','$scope', '
         $scope.panels = panels
 
         if (!$scope.view_tab && panels.length > 0 && $scope.panels[$scope.panels.length - 1].name.startsWith('intro')) {
```
Comment (kwatters) [inline src/main/resources/resource/WebGui/app/views/tabsViewCtrl.js:35]: No body in this if statement.. dead code.

File: src/test/java/org/myrobotlab/service/ArduinoTest.java [hunk 5]
```diff
@@ -483,15 +453,13 @@ public final void testServo() throws Exception {
     // notify and process releasing itself from attached
     // services
     servo.releaseService();
-
+    Service.sleep(200); // async call removal of servo
```
Comment (kwatters) [inline src/test/java/org/myrobotlab/service/ArduinoTest.java:456]: feels like a race condition here...

File: src/test/java/org/myrobotlab/service/NeoPixelTest.java [hunk 6]
```diff
@@ -5,30 +5,32 @@
 
 import static org.junit.Assert.assertTrue;
 
-import org.junit.Before;
+import org.junit.BeforeClass;
```
Comment (kwatters) [inline src/test/java/org/myrobotlab/service/NeoPixelTest.java:8]: why BeforeClass instead of just Before?

File: src/test/java/org/myrobotlab/service/ServoTest.java [hunk 7]
```diff
@@ -5,6 +5,7 @@
 import static org.junit.Assert.assertNull;
 import static org.junit.Assert.assertTrue;
 
+import org.junit.AfterClass;
```
Comment (kwatters) [inline src/test/java/org/myrobotlab/service/ServoTest.java:8]: Why after class instead of just After?

File: src/main/java/org/myrobotlab/service/Log.java (hunk 1 above)
Comment (supertick) [reply src/main/java/org/myrobotlab/service/Log.java:259]: required when implementing a Logger interface - ask jvm and jlog makers

File: src/main/java/org/myrobotlab/service/InMoov2Head.java (hunk 2 above)
Comment (supertick) [reply src/main/java/org/myrobotlab/service/InMoov2Head.java:479]: yeah gutted from within, for good reason - but should be re-implemented with sane implementation and config !

File: src/main/java/org/myrobotlab/service/Runtime.java (hunk 3 above)
Comment (supertick) [reply src/main/java/org/myrobotlab/service/Runtime.java:1463]: not sure .. but there's a lot of deps & code involved ... mebbe its only eclipse related and their remote tooling ... dunno

File: src/main/resources/resource/WebGui/app/views/tabsViewCtrl.js (hunk 4 above)
Comment (supertick) [reply src/main/resources/resource/WebGui/app/views/tabsViewCtrl.js:35]: heh ... you never go here :)
There's lots of bigger fish to fry in these javascript parts

File: src/test/java/org/myrobotlab/service/ArduinoTest.java (hunk 5 above)
Comment (supertick) [reply src/test/java/org/myrobotlab/service/ArduinoTest.java:456]: Arduino is a bit tough ... its attached in two realms java-land and mrlcom land
and they have to be sync'd
and there's connections and serial and ... and...
yeah still too much complexity ... but the "Java land is always right, mrlcomm needs to follow" is certainly helps
(but there are still details .... that are not all resolved)

File: src/test/java/org/myrobotlab/service/NeoPixelTest.java (hunk 6 above)
Comment (supertick) [reply src/test/java/org/myrobotlab/service/NeoPixelTest.java:8]: Before is before every test
BeforeClass is before all tests once

File: src/test/java/org/myrobotlab/service/ServoTest.java (hunk 7 above)
Comment (supertick) [reply src/test/java/org/myrobotlab/service/ServoTest.java:8]: same as above
